Sources/Views/Products/Detail/Summary/IngredientsAnalysisTableViewCell.swift

```diff
@@ -0,0 +1,88 @@
+//
+//  IngredientsAnalysisTableViewCell.swift
+//  OpenFoodFacts
+//
+//  Created by Timothee MATO on 14/10/2019.
+//  Copyright © 2019 Andrés Pizá Bückmann. All rights reserved.
+//
+
+import UIKit
+
+class IngredientsAnalysisTableViewCell: ProductDetailBaseCell {
+
+    @IBOutlet weak var palmOilView: IngredientsAnalysisView!
+    @IBOutlet weak var vegetarianView: IngredientsAnalysisView!
+    @IBOutlet weak var veganView: IngredientsAnalysisView!
+    var viewController: FormTableViewController?
+    var ingredientList: [Ingredient]?
+
+    override func configure(with formRow: FormRow, in viewController: FormTableViewController) {
+        guard let product = formRow.value as? Product else { return }
+
+        guard let ingredientsAnalysisDetails = product.ingredientsAnalysisDetails else { return }
+
+        ingredientList = product.ingredientsListAnalysis
+
+        for detail in ingredientsAnalysisDetails {
+            switch detail.type {
+            case IngredientsAnalysisType.palmOil:
+                palmOilView.configure(detail: detail)
+            case IngredientsAnalysisType.vegetarian:
+                vegetarianView.configure(detail: detail)
+            case IngredientsAnalysisType.vegan:
+                veganView.configure(detail: detail)
+            default:
+                break
+            }
+        }
+
+        palmOilView.addGestureRecognizer(UITapGestureRecognizer(target: self, action: #selector(didTap)))
```

Comment:
I think manually added gesture recognizer should be removed when dismissing to avoid memory leaks. To check.

Reply:
In viewWillDisappear?

Reply:
As it is a tableViewCell, gesture recognizers will be added each time the configure method is called, which is in the getCell method, therefore if the cell is reused by the table view, multiple gestures recognizers will be created. The best place to remove them is in the `tableView:didEndDisplayingCell:forRowAtIndexPath` (poorly named method, see  https://developer.apple.com/documentation/uikit/uitableviewdelegate/1614870-tableview?language=objc), so that gesture recognizers are removed immediatly when the cell is removed from the table view.

Reply:
Thanks, great advice.

Reply:
Beware to remove only the gesture that you added:
`        if let gestures = palmOilView?.gestureRecognizers {
            for gesture in gestures {
                if let tapGesture = gesture as? UITapGestureRecognizer,
                    tapGesture.numberOfTouchesRequired == 1 {
                    palmOilView?.removeGestureRecognizer(gesture)
                }
            }
        }
`

Reply:
comparing a gesture using only one prop (here the number of touches required) is kind of hacky. We could just keep a reference to the gesture recognizer in the cell and remove it, so we are sure we handle our gesture recognizers

Reply:
Agree, can a tag be added to a gesture?

Reply:
Not sure, but we could, instead of 

```swift
palmOilView.addGestureRecognizer(UITapGestureRecognizer(target: self, action: #selector(didTap)))
```

do this :  

```swift
self.palmOilGesture = UITapGestureRecognizer(target: self, action: #selector(didTap))
palmOilView.addGestureRecognizer(self.palmOilGesture)
```

and later just 
```swift
palmOilView.removeGestureRecognizer(self.palmOilGesture)
self.palmOilGesture = nil
```

Reply:
ok here, such a solution is unfortunately not always possible.

Reply:
Do you need to nillify as well?

Reply:
It's kind of a reflex, I always have a doubt on that kind of things so I add it to be sure I dont retain anything not needed